gateway/src/main/java/io/zeebe/gateway/impl/broker/BrokerRequestManager.java

```diff
@@ -170,18 +187,28 @@ private static boolean shouldRetryRequest(final IncomingResponse response) {
       actor.runOnCompletion(
           responseFuture,
           (clientResponse, error) -> {
+            boolean wasDropped = false;
+
             try {
               if (error == null) {
                 final BrokerResponse<T> response = request.getResponse(clientResponse);
                 responseConsumer.accept(response, null);
               } else {
                 responseConsumer.accept(null, error);
+                wasDropped = (error instanceof RequestTimeoutException);
               }
             } catch (final RuntimeException e) {
               responseConsumer.accept(null, new ClientResponseException(e));
+            } finally {
+              if (wasDropped) {
+                limiterListener.onDropped();
```

Comment:
I think dropped should also occur when you get exceptions that are unexpected, if I understood correctly - i.e. could not reach upstream should mark it as dropped, according to the docs. Do we want to do this here? I think at the moment we wouldn't, it would be caught by `RuntimeException`, and `wasDropped` would remain false, or...?

Reply:
Dropped requests results in updating limits. So I think in our case only Timeouts are relevant.